akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -40,8 +43,41 @@ object ActorTestKit {
   def apply(): ActorTestKit =
     new ActorTestKit(
       name = TestKitUtils.testNameFromCallStack(classOf[ActorTestKit]),
-      config = ApplicationTestConfig,
+      ActorSystemSetup.create(BootstrapSetup(ApplicationTestConfig)),
+      settings = None)
+
+  /**
+   * Create a testkit named based on the provided ActorSystem.
+   *
+   * It will create an [[akka.actor.typed.ActorSystem]] with this name,
+   * e.g. threads will include the name, and the related actor sustem setup.
+   * When the test has completed you should terminate the `ActorSystem` and
+   * the testkit with [[ActorTestKit#shutdownTestKit]].
+   *
+   * Config loaded from `system.settings.config` if that exists, otherwise
+   * using default configuration from the reference.conf resources that ship with the Akka libraries.
+   */
+  def apply(system: ActorSystem[_]): ActorTestKit = {
+    new ActorTestKit(
+      name = TestKitUtils.scrubActorSystemName(system.name),
+      ActorSystemSetup.create(BootstrapSetup(system.settings.config)),
       settings = None)
+  }
+
+  /**
+   * Create a testkit named based on the provided ActorSystemSetup.
+   *
+   * It will create an [[akka.actor.typed.ActorSystem]] with this name and actor system setup,
+   * e.g. threads will include the name and related actor system setup.
+   * When the test has completed you should terminate the `ActorSystem` and
+   * the testkit with [[ActorTestKit#shutdownTestKit]].
+   *
+   * Config loaded from `application-test.conf` if that exists, otherwise
+   * using default configuration from the reference.conf resources that ship with the Akka libraries.
+   */
+  def apply(name: String, actorSystemSetup: ActorSystemSetup): ActorTestKit = {
```

Comment:
If we have the support for passing in an ActorSystem as described above there might be less need of having this ActorSystemSetup variant since the user has full control of how the ActorSystem is created, including its Setup.